src/MOI_wrapper.jl

```diff
@@ -231,28 +251,57 @@ function MOI.optimize!(instance::Optimizer)
     b = instance.data.b
     G = ECOS.ECOSMatrix(sparse(instance.data.IG, instance.data.JG, instance.data.VG, m, n))
     h = instance.data.h
-    objconstant = instance.data.objconstant
+    objective_constant = instance.data.objective_constant
     c = instance.data.c
     instance.data = nothing # Allows GC to free instance.data before A is loaded to ECOS
+    options = instance.options
+    if instance.silent
+        options = copy(options)
+        options[:verbose] = false
+    end
     ecos_prob_ptr = ECOS.setup(n, m, cone.f, cone.l, length(cone.qa), cone.qa,
-                               cone.ep, G, A, c, h, b; instance.options...)
+                               cone.ep, G, A, c, h, b; options...)
+    start_time = time()
     ret_val = ECOS.solve(ecos_prob_ptr)
+    solve_time = time() - start_time
     ecos_prob = unsafe_wrap(Array, ecos_prob_ptr, 1)[1]
     primal    = unsafe_wrap(Array, ecos_prob.x, n)[:]
     dual_eq   = unsafe_wrap(Array, ecos_prob.y, cone.f)[:]
     dual_ineq = unsafe_wrap(Array, ecos_prob.z, m)[:]
     slack     = unsafe_wrap(Array, ecos_prob.s, m)[:]
     ECOS.cleanup(ecos_prob_ptr, 0)
-    objval = (instance.maxsense ? -1 : 1) * dot(c, primal)
-    if ret_val != ECOS.ECOS_DINF
-        objval += objconstant
-    end
-    objbnd = -(dot(b, dual_eq) + dot(h, dual_ineq))
-    if ret_val != ECOS.ECOS_PINF
-        objbnd += objconstant
+    objective_value = (instance.maxsense ? -1 : 1) * dot(c, primal)
+    dual_objective_value = (instance.maxsense ? 1 : -1) * (dot(b, dual_eq) + dot(h, dual_ineq))
+    instance.sol = Solution(ret_val, primal, dual_eq, dual_ineq, slack, objective_value,
+                            dual_objective_value, objective_constant, solve_time)
+end
+
+MOI.get(optimizer::Optimizer, ::MOI.SolveTime) = optimizer.sol.solve_time
+function MOI.get(optimizer::Optimizer, ::MOI.RawStatusString)
+    # Strings from https://github.com/ifa-ethz/ecos/blob/master/include/ecos.h
+    flag = optimizer.sol.ret_val
+    if flag == OPTIMIZE_NOT_CALLED
+        return "Optimize not called"
+    elseif flag == ECOS_OPTIMAL
+        return "Problem solved to optimality"
+    elseif flag == ECOS_PINF
+        return "Found certificate of primal infeasibility"
+    elseif flag == ECOS_DINF
+        return "Found certificate of dual infeasibility"
+    elseif flag == ECOS_INACC_OFFSET
```

Comment:
This is an incorrect interpretation of the flags. `ECOS_INACC_OFFSET` is a value that can be added to statuses to change the meaning. See https://github.com/JuliaOpt/ECOS.jl/blob/dbcf2e6ec707d6e3034bb4061e94f64d47cc29f1/src/MOI_wrapper.jl#L271.